src/GuzzleHandler.php

```diff
@@ -1,104 +1,112 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace LukeWaite\RingPhpGuzzleHandler;
 
-use GuzzleHttp\Client;
+use GuzzleHttp\ClientInterface;
 use GuzzleHttp\Exception\GuzzleException;
 use GuzzleHttp\RequestOptions;
 use GuzzleHttp\Ring\Core;
 use GuzzleHttp\Ring\Future\CompletedFutureArray;
 use GuzzleHttp\Ring\Future\FutureArrayInterface;
+use Psr\Http\Message\ResponseInterface;
 
 class GuzzleHandler
 {
     private $client;
 
-    public function __construct($client)
+    public function __construct(ClientInterface $client)
     {
         $this->client = $client;
     }
 
-    /**
-     * @param $request
-     * @return FutureArrayInterface
-     */
-    public function __invoke($request)
+    public function __invoke(array $request): FutureArrayInterface
     {
-        return new CompletedFutureArray(
-            $this->_invokeGuzzle($request)
-        );
+        return new CompletedFutureArray($this->invokeGuzzle($request));
     }
 
-    public function _invokeGuzzle($request)
+    private function invokeGuzzle(array $request): array
     {
         $url = Core::url($request);
         Core::doSleep($request);
 
         $stats = null;
+        $options = [
+            RequestOptions::BODY => Core::body($request),
+            RequestOptions::HEADERS => $request['headers'],
+            RequestOptions::HTTP_ERRORS => false,
+        ];
+
+        if (isset($request['client']['curl'][CURLOPT_USERPWD])) {
+            $options['auth'] = explode(':', $request['client']['curl'][CURLOPT_USERPWD]);
+        }
 
         try {
             $start = microtime(true);
-            $response = $this->client->request(
-                $request['http_method'],
-                $url,
-                [
-                    RequestOptions::BODY => Core::body($request),
-                    RequestOptions::HEADERS => $request['headers'],
-                    RequestOptions::HTTP_ERRORS => false
-                ]
-            );
+            $response = $this->client->request($request['http_method'], $url, $options);
             $end = microtime(true);
-        } catch (GuzzleException $e) {
-            return ['error' => $e];
+        } catch (GuzzleException $exception) {
+            return ['error' => $exception] + $this->emptyResponse();
```

Comment:
I merge the array here with the emptyResponse array because of https://github.com/elastic/elasticsearch-php/blob/562ec31a80a46157b9f5f87dc5b637d4d3486c5d/src/Elasticsearch/Connections/Connection.php#L257